### ISSUE TITLE
Out of order events break read receipts

### ISSUE THREAD

**Issue by dbkr** [vector-im/vector-web#789]
If an event comes in to a room that the server thinks belongs a way back in the graph and vector tries to send a read receipt for this graph, it will probably be ignored by the server because we have a receipt for a more recent message, meaning our read receipt will not advance to what vector thinks is the end of the room. Refreshing vector will fix the problem because vector will re-sync and the out-of-order event will go into its rightful place in the timeline.

This is due to the fact that the js sdk just appends all new events to the end of the timeline even if they were from some time ago.

**Comment by ara4n** [comment]
@dbkr: is this still an issue?

**Comment by ara4n** [comment]
yes, i'm assuming it still is, especially given empty events break RRs.

**Comment by dbkr** [comment]
I'm not sure if this is an issue any more. Since nobody else seems to have hit this, I'm not sure it makes sense to keep this open.
